### ISSUE TITLE
What about limitations on 'wait' & 'alias' command ?

### ISSUE THREAD

**Issue by Zetz512** [ValveSoftware/halflife#2523]
I will like if you has some cvars where you can control abuse from 'wait' & 'alias' commands like sv_extracmd_limit x (value in seconds) where you can use only 1 wait/alias every X seconds, of course, some innofensive scripts don't be affected for this control (like autobuy weapons, grenades, switch weapons ....) but scripts like autobunny, low recoil, autoduck, give you some advantage over others players and this can be disgusting for some players were are playing normaly this game...

If this isn't useful, atleast give some logs about what command use player with 'wait/alias' where you can see and take the control for himself

Or just disable the use of 'alias'/'wait' like sv_extra_allowed 0/1 (0 - disabled)

ty, sry for my bad english :9

**Comment by Maxi605** [comment]
alias is mostly used for cheating, i don't know if there's another use for it, probably there is.

**Comment by magister4813** [comment]
Those commands can't be filtered by hlds because they are strictly client-side commands.

**Comment by Zetz512** [comment]
Yep, i think that need change how alias & wait works and make a connection between client-side & server-side to check what you use and if this ok or not

**Comment by ptrBR** [comment]
Please @mikela-valve fix this, is so anoying players using this scripts and cheats with "alias" command.

**Comment by ChaosSpoofer** [comment]
The developers obviously saw fit to grant players access to aliases as well as the "wait" command.  Various communities like HL speedrunners, HLDMers, TFCers, etc., have fully adapted and embraced scripting and these commands at both the casual and competitive level, and eliminating them 20 years later would be disastrous.  The aforementioned communities would riot.

While Counter-Strike has always been an outlier as its community has regularly placed more stigma and restrictions on scripting, both aliases and the "wait" command have plenty of innocent and practical applications even in CS, short of those exploits you've described above.  If you want various exploits remedied, target the exploits, rather than the basic tools that make them (as well as plenty of perfectly innocent and accepted stuff throughout GoldSrc) possible.

**Comment by Zetz512** [comment]
Nice bro, your comment would make sense if i would suggest remove this completly from the game...

**Comment by Maxi605** [comment]
> The developers obviously saw fit to grant players access to aliases as well as the "wait" command. Various communities like HL speedrunners, HLDMers, TFCers, etc., have fully adapted and embraced scripting and these commands at both the casual and competitive level, and eliminating them 20 years later would be disastrous. The aforementioned communities would riot.
> 
> While Counter-Strike has always been an outlier as its community has regularly placed more stigma and restrictions on scripting, both aliases and the "wait" command have plenty of innocent and practical applications even in CS, short of those exploits you've described above. If you want various exploits remedied, target the exploits, rather than the basic tools that make them (as well as plenty of perfectly innocent and accepted stuff throughout GoldSrc) possible.

This only causes problems on the multiplayer community, alias shouldn't be usable on servers, do it on singleplayer, no problem, but servers? nah, the problem with cheating and other stuff thanks to alias is incredible, the anticheat is useless at this point, ANYTHING that can help fight back cheating and keep the game alive is welcome.

**Comment by SamVanheer** [comment]
Since alias and wait are both client side commands there is nothing that can be done to prevent their use. All a player would have to do is use an older/cracked/custom engine build to bypass any command restrictions. Now it's even more unfair because those players have a greater advantage over other players.

**Comment by Zetz512** [comment]
Mmmm, other thing i think can be useful is if you can "hook" the alias and "wait" commands for amxx developers to have more control about what you can use or not...

**Comment by SamVanheer** [comment]
Since they're client side commands they can't be hooked by server code, unless you enjoy having time sensitive commands be delayed by round trips. It would also block all commands that follow, including movement and weapon input.
